chartfx-math/src/main/java/de/gsi/math/MathBaseGenBase.java

```diff
@@ -0,0 +1,429 @@
+        return java.lang.Math.log(x);
+    }
+
+    public static double log10(double x) {
+        return java.lang.Math.log10(x);
+    }
+
+    public static double aCosH(double arg0) {
+        if (arg0 == 0.0)
+            return 0.0;
+        double ax = java.lang.Math.abs(arg0);
+        return java.lang.Math.log(arg0 + ax * java.lang.Math.sqrt(1. - 1. / (ax * ax)));
+    }
+
+    public static double aSinH(double arg0) {
+        if (arg0 == 0.0)
+            return 0.0;
+        double ax = java.lang.Math.abs(arg0);
+        return java.lang.Math.log(arg0 + ax * java.lang.Math.sqrt(1. + 1. / (ax * ax)));
+    }
+
+    public static double aTanH(double arg0) {
+        return java.lang.Math.log((1 + arg0) / (1 - arg0)) / 2;
+    }
+
+    // codegen: double -> long
+    /**
+     * some integer math
+     *
+     * @param x input px
+     * @param y input py
+     * @return sqrt(px*px + py*py)
+     */
+    public static double hypot(double x, double y) {
+        return java.lang.Math.hypot(x, y);
+    }
+    //// end codegen
+
+    //// codegen: double -> float, int, long
+    public static double abs(double d) {
+        return (d >= 0) ? d : -d;
+        // short: return (d >= 0) ? (short) d : (short) (-d);
+    }
+    //// end codegen
+
+    public static double aCos(double arg0) {
+        return java.lang.Math.acos(arg0);
+    }
+
+    public static double aSin(double arg0) {
+        return java.lang.Math.asin(arg0);
+    }
+
+    public static double aTan(double arg0) {
+        return java.lang.Math.atan(arg0);
+    }
+
+    public static double aTan2(double y, double x) {
+        if (x != 0)
+            return java.lang.Math.atan2(y, x);
+        else if (y == 0)
+            return 0;
+        else if (y > 0)
+            return PI / 2;
+        else
+            return -PI / 2;
+    }
+
+    public static double ceil(double x) {
+        return java.lang.Math.ceil(x);
+    }
+
+    public static int ceilNInt(double x) {
+        return nInt(java.lang.Math.ceil(x));
+    }
+
+    public static double cos(double arg0) {
+        return java.lang.Math.cos(arg0);
+    }
+
+    public static double cosH(double arg0) {
+        return java.lang.Math.cosh(arg0);
+    }
+
+    public static boolean even(long a) {
+        return !((a & 1) != 0);
+    }
+
+    public static double exp(double x) {
+        return java.lang.Math.exp(x);
+    }
+
+    public static boolean finite(double x) {
+        return !(Double.isInfinite(x) || Double.isNaN(x));
+    }
+
+    public static double floor(double x) {
+        return java.lang.Math.floor(x);
+    }
+
+    public static int floorNint(double x) {
+        return nInt(java.lang.Math.floor(x));
+    }
+
+    public static boolean isNaN(double x) {
+        return Double.isNaN(x);
+    }
+
+    public static double ldExp(double x, int exp) {
+        return x * java.lang.Math.pow(2, exp);
+    }
+
+    public static double log2(double x) {
+        return java.lang.Math.log(x) / LOG_2;
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'java.lang.Math.log' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)